sdc/datatypes/hpat_pandas_stringmethods_functions.py

```diff
@@ -1233,6 +1123,12 @@ def hpat_pandas_stringmethods_strip(self, to_strip=None):
         'seealso': seealso_strip_methods,
         'limitations': ''
     },
+    'lower': {
+        'method': hpat_pandas_stringmethods_lower,
+        'caption': 'Convert strings in the Series to lowercase.',
+        'seealso': seealso_transform_methods,
+        'limitations': ''
```

Comment:
@densmirn @Rubtsowa are you sure there is no limitation?
As ar as I remember we are not supporting None in strings

Reply:
@AlexanderKalistratov according to the changes in the PR, NaNs are supported in strings. So limitations were dropped from the documentation.

Reply:
@densmirn we are converting `None` into `Nan`. But pandas doesn't:
```
>>> import pandas
>>> import numpy
>>> s = pandas.Series(["A", "B", "C", None, numpy.nan])
>>> print(s)
0       A
1       B
2       C
3    None
4     NaN
dtype: object
>>> s1 = s.str.lower()
>>> print(s1)
0       a
1       b
2       c
3    None
4     NaN
dtype: object 
```

On other hand, pandas assume `None` and `Nan` as the same value:
```
>>> s2 = pandas.Series(["A", None, numpy.nan])
>>> s3 = pandas.Series(["A", None, None])
>>> pandas.testing.assert_series_equal(s2, s3)
>>> b = pandas.testing.assert_series_equal(s2, s3)
```

Reply:
Perhaps it makes sense to add it as a limitation. @Rubtsowa could you do that?

Reply:
If so, it should be added to all string methods

Reply:
Of course, but let's make such change for other methods in separate PR.